featuretools/computational_backends/utils.py

```diff
@@ -198,3 +205,72 @@ def get_client_cluster():
     return Client, LocalCluster
+
+
+def _validate_cutoff_time(cutoff_time, target_entity):
+    """
+    Verify that the cutoff time is a single value or a pandas dataframe with the proper columns
+    containing no duplicate rows
+    """
+    if isinstance(cutoff_time, dd.DataFrame):
+        msg = "cutoff_time should be a Pandas DataFrame: "\
+            "computing cutoff_time, this may take a while"
+        warnings.warn(msg)
+        cutoff_time = cutoff_time.compute()
+
+    if isinstance(cutoff_time, pd.DataFrame):
+        cutoff_time = cutoff_time.reset_index(drop=True)
+
+        if "instance_id" not in cutoff_time.columns:
+            if target_entity.index not in cutoff_time.columns:
+                raise AttributeError('Cutoff time DataFrame must contain a column with either the same name'
+                                     ' as the target entity index or a column named "instance_id"')
+            # rename to instance_id
+            cutoff_time.rename(columns={target_entity.index: "instance_id"}, inplace=True)
+
+        if "time" not in cutoff_time.columns:
+            if target_entity.time_index and target_entity.time_index not in cutoff_time.columns:
+                raise AttributeError('Cutoff time DataFrame must contain a column with either the same name'
+                                     ' as the target entity time_index or a column named "time"')
+            # rename to time
+            cutoff_time.rename(columns={target_entity.time_index: "time"}, inplace=True)
+
+        # Make sure user supplies only one valid name for instance id and time columns
+        if "instance_id" in cutoff_time.columns and target_entity.index in cutoff_time.columns and \
+                "instance_id" != target_entity.index:
+            raise AttributeError('Cutoff time DataFrame cannot contain both a column named "instance_id" and a column'
+                                 ' with the same name as the target entity index')
+        if "time" in cutoff_time.columns and target_entity.time_index in cutoff_time.columns and \
+                "time" != target_entity.time_index:
+            raise AttributeError('Cutoff time DataFrame cannot contain both a column named "time" and a column'
+                                 ' with the same name as the target entity time index')
+
+        assert (cutoff_time[['instance_id', 'time']].duplicated().sum() == 0), \
+            "Duplicated rows in cutoff time dataframe."
+    else:
+        if isinstance(cutoff_time, list):
+            raise TypeError("cutoff_time must be a single value or DataFrame")
+
+    return cutoff_time
+
+
+def _check_cutoff_time_type(cutoff_time, es_time_type):
+    """
+    Check that the cutoff time values are of the proper type given the entityset time type
+    """
+    # Check that cutoff_time time type matches entityset time type
+    if isinstance(cutoff_time, tuple):
+        cutoff_time_dtype = type(cutoff_time[0])
+        numeric_types = [int, float]
```

Comment:
We may want to add more numeric types here.  Things like `cutoff_time=pd.Series([1])[0]` will cause trouble here, and the error message and using pd.to_numeric would be unhelpful currently

Reply:
the `_check_time_type` logic checks for numpy variations

Reply:
Changed the strategy to check if the cutoff time is an instance of `numbers.Number` which should take care of the scenario you mentioned where the type is `numpy.int64` or something along those lines.

Do you see any issues with this new approach?

Reply:
I missed your comment before on `_check_time_type`. I updated the code to use `_check_time_type` for single cutoff time values instead of the approach with `numbers.Number` - leveraging existing code and eliminating additional imports.